### ISSUE TITLE
Accept multiple tracks as parameters to the "fetch" endpoint.

### ISSUE THREAD

**Issue by kytrinyx** [exercism/x-api#84]
Right now we have `/v2/exercises` and `/v2/exercises/:track_id`. Let's tweak the `/v2/exercises` to take an optional parameter `?tracks=a,b,c` and if that parameter is passed, only return the data for those tracks.

**Comment by kytrinyx** [comment]
Reserved by @ben-biddington

**Comment by ben-biddington** [comment]
Apologies for the donut question, but I am getting approval errors like:

```
$ bundle ex ruby test/v3/routes/tracks_test.rb 
Run options: --seed 51013

# Running:

.E

Fabulous run in 0.034198s, 58.4832 runs/s, 204.6913 assertions/s.

  1) Error:
V3RoutesTracksTest#test_tracks:
Approvals::ApprovalError: Approval Error: Received file "./test/fixtures/approvals/v3_track.received.txt" does not match approved "./test/fixtures/approvals/v3_track.approved.txt".
    /home/ben/.rvm/gems/ruby-2.2.1/gems/approvals-0.0.22/lib/approvals/approval.rb:85:in `fail_with'
    /home/ben/.rvm/gems/ruby-2.2.1/gems/approvals-0.0.22/lib/approvals/approval.rb:49:in `verify'
    /home/ben/.rvm/gems/ruby-2.2.1/gems/approvals-0.0.22/lib/approvals/dsl.rb:4:in `verify'
    test/v3/routes/tracks_test.rb:24:in `test_tracks'

2 runs, 7 assertions, 0 failures, 1 errors, 0 skips
Coverage report generated for Unit Tests to /home/ben/sauce/x-api/coverage. 353 / 497 LOC (71.03%) covered.
```
```
diff "./test/fixtures/approvals/v3_track.received.txt" "./test/fixtures/approvals/v3_track.approved.txt"
5c5
< 	"todo" => ["cherry", "banana", "apple", "dog"]
---
> 	"todo" => ["apple", "banana", "cherry", "dog"]

```

Any ideas?

I have these versions:

```
$ git show 
commit d84cb6e9556392f0a7fcadc5bde9b3d352ba7a74
Merge: d9784fc 67c0aee
Author: Katrina Owen <katrina.owen@gmail.com>
Date:   Thu Dec 3 17:36:25 2015 -0700

    Merge pull request #87 from GustavoCaso/handled_errors_with_tests_and_readme_endpoints
    
    Added status 404 to tests and readme endpoint
```
```
$ ruby -v 
ruby 2.2.1p85 (2015-02-26 revision 49769) [x86_64-linux]
```
```
$ bundle show 
Gems included by the bundle:
  * addressable (2.3.8)
  * approvals (0.0.22)
  * ast (2.1.0)
  * astrolabe (1.3.1)
  * bugsnag (2.8.12)
  * bundler (1.10.6)
  * coveralls (0.8.9)
  * crack (0.4.2)
  * docile (1.1.5)
  * domain_name (0.5.25)
  * faraday (0.9.2)
  * http-cookie (1.0.2)
  * json (1.8.3)
  * mime-types (2.99)
  * mini_portile (0.6.2)
  * minitest (5.8.3)
  * multipart-post (2.0.0)
  * netrc (0.11.0)
  * nokogiri (1.6.6.4)
  * org-ruby (0.9.12)
  * parser (2.2.3.0)
  * petroglyph (0.0.7)
  * powerpack (0.1.1)
  * psych (2.0.15)
  * puma (2.15.3)
  * rack (1.6.4)
  * rack-protection (1.5.3)
  * rack-test (0.6.3)
  * rainbow (2.0.0)
  * rake (10.4.2)
  * rest-client (1.8.0)
  * rubocop (0.30.1)
  * ruby-progressbar (1.7.5)
  * rubypants (0.2.0)
  * safe_yaml (1.0.4)
  * simplecov (0.10.0)
  * simplecov-html (0.10.0)
  * sinatra (1.4.6)
  * term-ansicolor (1.3.2)
  * thor (0.19.1)
  * tilt (1.4.1)
  * tins (1.6.0)
  * unf (0.1.4)
  * unf_ext (0.0.7.1)
  * vcr (2.9.3)
  * webmock (1.17.4)
```
```
$ date 
Fri Dec  4 20:57:39 NZDT 2015
```

**Comment by kytrinyx** [comment]
Weird. I wonder if something was sorting implicitly, and needs to be explicit?

**Comment by ben-biddington** [comment]
Tis, when I run [this](https://github.com/ben-biddington/x-api/blob/%2384/test/v1/routes/tracks_test.rb#L34) (line 37 has diagnostic output) locally I get:
```
ACTUAL: ["cherry", "banana", "apple", "dog"], EXPECTED: ["apple", "banana", "cherry", "dog"]
```

But [in CI](https://travis-ci.org/ben-biddington/x-api/builds/95448519#L549):

```
ACTUAL: ["apple", "banana", "cherry", "dog"], EXPECTED: ["apple", "banana", "cherry", "dog"]
```

Have tried aligning exact versions of rubygems and bundler, still no good.

**Comment by kytrinyx** [comment]
How about sorting the "todo" key in the response body of the request (or earlier, for that matter) so that we don't rely on implicit behavior?

**Comment by ben-biddington** [comment]
Could do -- I just wonder what makes it differ

**Comment by kytrinyx** [comment]
My guess is that it has to do with the order of directories when reading them off the filesystem. That's usually what has me running into weird issues on different platforms.

**Comment by ben-biddington** [comment]
Ooh yes! One sec

**Comment by ben-biddington** [comment]
Looks like that's it. Below is output from `Dir["#{dir}/*.yml"]`

local:

```
["./test/fixtures/metadata/cherry.yml", 
 "./test/fixtures/metadata/one.yml",
 "./test/fixtures/metadata/two.yml", 
 "./test/fixtures/metadata/three.yml",
 "./test/fixtures/metadata/hello-world.yml", 
 "./test/fixtures/metadata/banana.yml",
 "./test/fixtures/metadata/apple.yml",   
 "./test/fixtures/metadata/dog.yml"]
```
CI:

```
["./test/fixtures/metadata/apple.yml", 
 "./test/fixtures/metadata/banana.yml", 
 "./test/fixtures/metadata/cherry.yml",
 "./test/fixtures/metadata/dog.yml", 
 "./test/fixtures/metadata/hello-world.yml", 
 "./test/fixtures/metadata/one.yml", 
 "./test/fixtures/metadata/three.yml", 
 "./test/fixtures/metadata/two.yml"]
```

Will submit change

**Comment by ben-biddington** [comment]
:green_apple: All green after #89

**Comment by ben-biddington** [comment]
@kytrinyx, did you mean to update API v2 or add this to API v3?

**Comment by kytrinyx** [comment]
I meant to update API v2--we're going to tackle v3 a bit differently.

**Comment by kytrinyx** [comment]
Closed in #90
